homeassistant/components/nibe_heatpump/config_flow.py

```diff
@@ -98,9 +121,55 @@ async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str,
     finally:
         await connection.stop()
 
-    return {
-        "title": f"{data[CONF_MODEL]} at {data[CONF_IP_ADDRESS]}",
+    return f"{data[CONF_MODEL]} at {data[CONF_IP_ADDRESS]}", {
+        **data,
         CONF_WORD_SWAP: word_swap,
+        CONF_CONNECTION_TYPE: CONF_CONNECTION_TYPE_NIBEGW,
+    }
+
+
+async def validate_modbus_input(
+    hass: HomeAssistant, data: dict[str, Any]
+) -> tuple[str, dict[str, Any]]:
+    """Validate the user input allows us to connect."""
+
+    heatpump = HeatPump(Model[data[CONF_MODEL]])
+    await heatpump.initialize()
+
+    try:
+        connection = Modbus(
+            heatpump,
+            data[CONF_MODBUS_URL],
+            data[CONF_MODBUS_UNIT],
+        )
+    except ValueError as exc:
+        raise FieldError("Not a valid modbus url", CONF_MODBUS_URL, "url") from exc
+
+    await connection.start()
+
+    try:
+        coil = heatpump.get_coil_by_name("reset-alarm-40023")
+        coil = await connection.read_coil(coil)
+        value: str | int = 0
+        if coil.mappings:
+            value = coil.mappings[str(value)]
+        coil.value = value
+        coil = await connection.write_coil(coil)
+    except CoilNotFoundException as exception:
+        raise FieldError(
+            "Model selected doesn't seem to support expected coils", "base", "model"
+        ) from exception
+    except CoilReadException as exception:
+        raise FieldError("Timeout on read from pump", "base", "read") from exception
+    except CoilWriteException as exception:
+        raise FieldError("Timeout on writing to pump", "base", "write") from exception
+    finally:
+        await connection.stop()
```

Comment:
It would be nicer (not required) if some of this was moved to the library to reduce the amount of code in HA

Reply:
Yes, we could have some verify connection function in lib. That would indeed make more sense. Will look at that.

Reply:
The exception mapping have to remain. But i've moved the verification code.